File: android/app/src/main/java/com/dailymemory/domain/usecase/reminder/SaveReminderUseCase.kt
package com.dailymemory.domain.usecase.reminder

import com.dailymemory.domain.model.Reminder
import com.dailymemory.domain.repository.ReminderRepository
import javax.inject.Inject

/**
 * Use case for saving a new reminder.
 */
class SaveReminderUseCase @Inject constructor(
    private val reminderRepository: ReminderRepository
) {
    suspend operator fun invoke(reminder: Reminder): Result<Reminder> {
        return try {
            reminderRepository.insert(reminder)
            Result.success(reminder)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
}

File: ios/DailyMemory/Domain/UseCase/Reminder/SaveReminderUseCase.swift
import Foundation

/// Use case for saving a new reminder.
final class SaveReminderUseCase {
    private let reminderRepository: ReminderRepository

    init(reminderRepository: ReminderRepository = ReminderRepositoryImpl()) {
        self.reminderRepository = reminderRepository
    }

    func execute(_ reminder: Reminder) async throws -> Reminder {
        try await reminderRepository.save(reminder)
        return reminder
    }
}
